packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -512,4 +533,12 @@ class _OutlineBorder extends ShapeBorder {
 
   @override
   int get hashCode => hashValues(side, shape);
+
+  @override
+  ShapeBorder resolve(Set<MaterialState> states) {
```

Comment:
I'm not sure I understand the need for the ShapeBorder to be a MaterialStateProperty. Could the color resolving not happen inside the `_getOutline` function? Maybe I'm misunderstanding.

Reply:
The Color resolving can't happen in `_getOutline` because `OutlineButton` doesn't actually know about it's focus/hover state, `RawMaterialButton` does. So rather than having `RawMaterialButton` do a check for `OutlineBorder` (which would also have to be made public), I figured it's better to use the `MaterialStateProperty` interface.